### ISSUE TITLE
Purpose of padding at dataloader.py

### ISSUE THREAD

**Issue by akhanss** [yhenon/pytorch-retinanet#32]
I am just wondering why we're using padding in Resizer() function at dataloader.py.
....
        pad_w = 32 - rows%32
        pad_h = 32 - cols%32

        new_image = np.zeros((rows + pad_w, cols + pad_h, cns)).astype(np.float32)
        new_image[:rows, :cols, :] = image.astype(np.float32)
....

**Comment by yhenon** [comment]
It's necessary for the image width/height to be a multiple of 32 because of the downsampling+upsampling in the network.

If the images are not, the sizes are rounded when the downsampling layers are applied. Then when we upsample, the original image size is not recovered, which causes issues.
